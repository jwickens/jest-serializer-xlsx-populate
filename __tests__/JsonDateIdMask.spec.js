
const randomNumber = () => Math.floor((Math.random() * 100));

describe('jest-serializer-json-date-id-mask', () => {
  it('wont do anything to non objects', () => {
    expect('id').toMatchSnapshot()
    expect(new Date('1990')).toMatchSnapshot()
    expect(0).toMatchSnapshot()
  })

  it('it will serialize ids', () => {
    expect({myId: randomNumber()}).toMatchSnapshot()
  })

  it('it will serialize todays date', () => {
    expect({someKey: new Date(), someOtherKey: (new Date()).toISOString()}).toMatchSnapshot()
  })

  it('works with nested objects', () => {
    expect({id: {id: {date: new Date(), id: randomNumber(), someOtherDate: (new Date()).toLocaleDateString()}}}).toMatchSnapshot()
  })
});