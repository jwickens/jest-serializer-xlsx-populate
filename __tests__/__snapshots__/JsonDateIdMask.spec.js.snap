// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`jest-serializer-json-date-id-mask it will serialize ids 1`] = `
{
  "myId": "<id>"
}
`;

exports[`jest-serializer-json-date-id-mask it will serialize todays date 1`] = `
{
  "someKey": "<date>",
  "someOtherKey": "<date>"
}
`;

exports[`jest-serializer-json-date-id-mask wont do anything to non objects 1`] = `"id"`;

exports[`jest-serializer-json-date-id-mask wont do anything to non objects 2`] = `"1990-01-01T00:00:00.000Z"`;

exports[`jest-serializer-json-date-id-mask wont do anything to non objects 3`] = `0`;

exports[`jest-serializer-json-date-id-mask works with nested objects 1`] = `
{
  "id": {
    "id": {
      "date": "<date>",
      "id": "<id>",
      "someOtherDate": "<date>"
    }
  }
}
`;
